cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/goals/RackAwareCapacityGoal.java

```diff
@@ -204,7 +206,17 @@ protected void initGoalState(ClusterModel clusterModel)
       }
     }
     // Sanity Check #3 -- i.e. not enough racks to satisfy rack awareness.
-    if (clusterModel.maxReplicationFactor() > clusterModel.numHealthyRacks()) {
+    int maxReplicationFactorOfIncludedTopics = 1;
+    Map<String, Integer> replicationFactorByTopic = clusterModel.replicationFactorByTopic();
+
+    for (Map.Entry<String, Integer> replicationFactorByTopicEntry: replicationFactorByTopic.entrySet()) {
+      if (!excludedTopics.contains(replicationFactorByTopicEntry.getKey())) {
+        maxReplicationFactorOfIncludedTopics =
```

Comment:
Exception could be thrown immediately if the replication factor is larger than the rack number. There is no need to iterate over all the topics anymore.